main/Main.hs

```diff
@@ -45,14 +55,30 @@ isServerVersionSupported = do
       HE.unit (HD.singleRow $ HD.value HD.int4) False
 
 {-|
+  The purpose of this worker is to fill the refDbStructure created in 'main'
+  with the 'DbStructure' returned from calling 'getDbStructure'. This method
+  is meant to be called by multiple times by the same thead, but does nothing if
+  the previous invocation has not terminated. In all cases this method does not
+  halt the caling thead, the work is proformed in a separate thread.
+
+  Note: 'atomicWriteIORef' is essentialy a lazy semaphore that prevents two
```

Comment:
essentialy -> essentially

Does your editor have a spell checker? I usually run my code/docs through one just in case, since it helps catch little typos.